cli/src/test/java/hudson/cli/PrivateKeyProviderTest.java

```diff
@@ -0,0 +1,79 @@
+package hudson.cli;
+
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.IllegalArgumentException;
+import java.security.GeneralSecurityException;
+import java.security.KeyPair;
+import java.security.NoSuchAlgorithmException;
+
+import static org.junit.Assert.assertNotNull;
+
+public class PrivateKeyProviderTest {
+
+    @Test
+    public void loadKeyDSA() throws IOException, GeneralSecurityException {
+        File file = new File(this.getClass().getResource("dsa").getFile());
+        String password = null;
+        KeyPair keyPair = PrivateKeyProvider.loadKey(file, password);
+        assertNotNull(keyPair);
+        assertNotNull(keyPair.getPrivate());
+        assertNotNull(keyPair.getPublic());
+    }
+
+    @Test
+    public void loadKeyDSAPassword() throws IOException, GeneralSecurityException {
+        File file = new File(this.getClass().getResource("dsa-password").getFile());
+        String password = "password";
+        KeyPair keyPair = PrivateKeyProvider.loadKey(file, password);
+        assertNotNull(keyPair);
+        assertNotNull(keyPair.getPrivate());
+        assertNotNull(keyPair.getPublic());
+    }
+    
+    @Test
+    public void loadKeyRSA() throws IOException, GeneralSecurityException {
+        File file = new File(this.getClass().getResource("rsa").getFile());
+        String password = null;
+        KeyPair keyPair = PrivateKeyProvider.loadKey(file, password);
+        assertNotNull(keyPair);
+        assertNotNull(keyPair.getPrivate());
+        assertNotNull(keyPair.getPublic());
+    }
+
+    @Test
+    public void loadKeyRSAPassword() throws IOException, GeneralSecurityException {
+        File file = new File(this.getClass().getResource("rsa-password").getFile());
+        String password = "password";
+        KeyPair keyPair = PrivateKeyProvider.loadKey(file, password);
+        assertNotNull(keyPair);
+        assertNotNull(keyPair.getPrivate());
+        assertNotNull(keyPair.getPublic());
+    }
+
+    @Test
+    public void loadKeyOpenSSH() throws IOException, GeneralSecurityException {
+        File file = new File(this.getClass().getResource("openssh").getFile());
+        String password = null;
+        KeyPair keyPair = PrivateKeyProvider.loadKey(file, password);
+        assertNotNull(keyPair);
+        assertNotNull(keyPair.getPrivate());
+        assertNotNull(keyPair.getPublic());
+    }
+
+    @Test(expected = NoSuchAlgorithmException.class)
+    public void loadKeyUnsupportedCipher() throws IOException, GeneralSecurityException {
```

Comment:
Could you add a comment explaining how the key was generated, what cypher is not supported, etc. Without more information the 3 keys are equivalent

Reply:
sure